src/Apps/CheezeWizards/Services/Firebase/FirebaseService.js

```diff
@@ -0,0 +1,239 @@
+            return db
+                .collection('wizards')
+                .doc('network')
+                .collection(network)
+                .doc(wizard.id)
+                .set(wizard, {
+                    merge: true
+                });
+        }));
+    }
+
+    // Todo: should handle network not being defined or invalid + associated test
+    // Todo: should check params passed in valid + associated test
+    // Todo: do we need to do ownership check that we're challenging someone else's wizard?
+    async sendChallenge(network, {challengeId, challengingWizardId, otherWizardId}) {
+
+        checkValidNetwork(network);
+
+        // challenger data
+        const challengerData = {
+            currentDuel: '0x0000000000000000000000000000000000000000000000000000000000000000',
+            otherWizardId,
+            challengeAccepted: false,
+            challenger: true,
+        };
+
+        // challengee data
+        const challengeeData = {
+            currentDuel: '0x0000000000000000000000000000000000000000000000000000000000000000',
+            challengingWizardId,
+            commitmentHash: '',
+            challengeAccepted: false,
+            challenger: false,
+        };
+
+        const networkRef = db.collection('wizards').doc('network').collection(network);
+        const challengerDataRef = networkRef.doc(challengingWizardId).collection('duel').doc(challengeId);
+        const challengeeDataRef = networkRef.doc(otherWizardId).collection('duel').doc(challengeId);
+
+        await db.runTransaction(t => {
+            t.set(challengerDataRef, challengerData);
+            t.set(challengeeDataRef, challengeeData);
+            return Promise.resolve('done');
+        });
+    }
+
+    async acceptChallenge(network, {wizardId, challengeId, commitmentHash, currentDuel}) {
+
+        checkValidNetwork(network);
+
+        const networkRef = db.collection('wizards').doc('network').collection(network);
+        const challengeRef = networkRef
+            .doc(wizardId)
+            .collection('duel')
+            .doc(challengeId);
+
+        await db.runTransaction(t => {
+           return t.get(challengeRef).then(doc => {
+               const challenge = doc.data();
+               const challengerRef = networkRef.doc(challenge.challengingWizardId).collection('duel').doc(challengeId);
+               return t.get(challengerRef).then(() => {
+                   t.update(challengeRef, {
+                       challengeAccepted: true,
+                       commitmentHash,
+                       currentDuel
+                   });
+
+                   t.update(challengerRef, {challengeAccepted: true});
+               });
+           });
+        });
+    }
+
+    async allOutstandingChallenges(network, owner) {
+        // TODO I need this in the stage
+    }
+
+    async registerFirebaseMessagingTokenForEthAddress() {
+
+    }
+
+    // async registerWizardForDueling(network, {owner, wizard}) {
```

Comment:
Ditch old code which is no longer needed

Reply:
I'm not sure about the comments - will need to speak with Andy about them